### PR TITLE
Security middleware

### PR DESCRIPTION
**Pull request by ktruedat**

### CODE REVIEW

**Comment by madalv** [comment]
Access token isn't saved in Redis on login? Why is pb folder named security_pb?

**Comment by ktruedat** [comment]
> Access token isn't saved in Redis on login? Why is pb folder named security_pb?

Why storing the access tokens in redis if we do not retrieve them in any way, we only work with refresh tokens.
security_pb will be renamed

File: gateway/internal/middleware/authorization.go
Comment (auperman-lab) [inline gateway/internal/middleware/authorization.go:1]: change error messages

File: security_service/internal/proto/security_msg.proto [hunk 1]
```diff
@@ -1,36 +1,43 @@
...
-
 message Empty {
 }
 
 message UserInfo {
-    uint64 id = 1;
-    string email = 2;
-    string password = 3;
-    string role = 4;
+  uint64 id = 1;
```
Comment (auperman-lab) [inline security_service/internal/proto/security_msg.proto:10]: change everywhere to uint32

File: gateway/internal/middleware/authorization.go
Comment (ktruedat) [reply gateway/internal/middleware/authorization.go:1]: ?

File: gateway/internal/middleware/authorization.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	return &AuthMiddleware{userClient: userClient, cfg: rbacCfg}
+}
+
+func (auth *AuthMiddleware) Authorize(action, resourceCode string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		mail, ok := c.Get("userEmail")
+		if !ok {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{
+				"error":   "Unauthorized",
+				"message": "You do not have permission to perform this action.",
```
Comment (auperman-lab) [inline gateway/internal/middleware/authorization.go:26]: repetitive error I mean

Comment (ktruedat) [reply gateway/internal/middleware/authorization.go:26]: What do you mean repetetive error? If there is no email in the context why should i let the request through?

**Pull request merged**